### DISCUSSION TITLE
Monitoring UDP drops causing pubsub to breakdown?

### DISCUSSION THREAD

**Discussion by chengguizi** [eclipse-ecal/ecal#1518] [Q&A]
Hi eCAL team,

We have been using eCAL to send relatively high bandwidth data over LAN. There is an issue affecting smooth operation.

When we are sending a coupld of image streams concurrently, say the bandwidth is about 20MB/s, we notice the ecal Monitor at host machine will lose the publisher node on the edge machine.

I suspect this is due to the fact that, monitoring is transmitted using UDP, and natually it is unreliable, especially when the network is under load.

May question would be:
- What is the current behaviour, when eCAL subscriber node stop receiving ecal monitoring heartbeat, but UDP / TCP messages continuously to arrive?
- In other word, once the TCP / UDP sub node is intialised with first message received, will its transmission affected by failure of receiving eCAL monitoring heartbeat within the timeout duration?

Thanks!

**Comment by rex-schilasky** [comment]
Hi @chengguizi,

in general, it is not optimal to load the network so much that a large number of messages are lost. The eCAL registration (UDP) layer is sending heartbeat information with a default frequency of 1 Hz. See `ecal.ini` file

![image](https://github.com/eclipse-ecal/ecal/assets/49162693/50763593-af0c-466b-be56-0abd6dcb1180)

The monitor is collecting these information to monitor the whole eCAL system. This debug visualization has its own timeout, default setting is 5 sec. See `ecal.ini`

![image](https://github.com/eclipse-ecal/ecal/assets/49162693/1523e33a-4899-48f8-bb3f-7fe21e033c4f)

Publisher start sending payloads right after the first receive of a matching registration heartbeat and stop sending after a timeout that is configured in `ecal.ini` as well (default 60 sec), see

![image](https://github.com/eclipse-ecal/ecal/assets/49162693/cbae60dd-8e67-48c0-85b9-a9589920ca12)

So, the publisher/subscriber may disappear in the monitoring visualization but transmitting payloads. But if the final registration timeout is reached they will stop sending until the get a new heartbeat message from their counterparts.

**Reply by chengguizi** [reply]
Thanks a lot for the insight! We will try to tune accordingly.
